core/src/main/java/io/questdb/griffin/WhereClauseParser.java

```diff
@@ -1158,6 +1181,64 @@ private boolean removeWithin(AliasTranslator translator, ExpressionNode node, Re
         }
     }
 
+    private void processArgument(
+            ExpressionNode inArg,
+            RecordMetadata metadata,
+            FunctionParser functionParser,
+            SqlExecutionContext executionContext,
+            int columnType,
+            LongList prefixes
+    ) throws SqlException, NumericException {
+        if(isNull(inArg)) {
+            throw SqlException.$(inArg.position, "GeoHash value expected");
+        }
+        if (isFunc(inArg)) {
+            Function f = functionParser.parseFunction(inArg, metadata, executionContext);
+            if (isGeoHashConstFunction(f)) {
+                final int fnType = f.getType();
+                final long hash = f.getGeoHashLong(null);
+                GeoHashes.addNormalizedGeoPrefix(hash, fnType, columnType, prefixes);
+            } else {
+                throw SqlException.$(inArg.position, "GeoHash const function expected");
+            }
+        } else {
+            final boolean isConstant = inArg.type == ExpressionNode.CONSTANT;
+            final int len = inArg.token.length();
+            final boolean isBitsPrefix = len > 2 && inArg.token.charAt(0) == '#' && inArg.token.charAt(1) == '#';
+            final boolean isCharsPrefix = len > 1 && inArg.token.charAt(0) == '#';
+
+            if (!(isConstant && (isBitsPrefix || isCharsPrefix))) {
+                throw SqlException.$(inArg.position, "GeoHash literal expected");
+            }
+
+            final int type;
+            final long hash;
+            if (isBitsPrefix) {
+                type = ColumnType.geohashWithPrecision(inArg.token.length() - 2);
```

Comment:
Can length be > 62 here? It will result in invalid type

Reply:
ExpressionParser will throw exception on invalid bits literal before this code.
addNormalizedGeoPrefix does prefix/column precision mismatch check later